PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/runner/utils/AppDeploySession.kt

```diff
@@ -66,6 +78,25 @@ class AppDeploySession(username: String, password: String) {
                 .post(requestBody)
                 .build()
 
-        client.newCall(request).execute().use { response -> return response }
+        client.newCall(request).execute().use { response ->
+            // If the initial async deploy fails, there is no use in polling.
+            if (!response.isSuccessful) return response
+
+            // If it succeeds, poll for a result
+            val pollingUrl = response.header("Location") ?: throw ExecutionException("The Azure KUDU Service did not return an URL to poll for deployment status.")
+            val maximumPollingDurationSeconds = (pollingTimeoutMs * pollingNumberOfTries) / 1000
+
+            return waitForResponse(pollingUrl, timeoutInSeconds = readTimeoutMs, client)
+                    .withRetries(pollingNumberOfTries,
```

Comment:
I'm confused here. So, do we wait for 30 minutes in total for publish web/function app? 3 attempts * 10 sec for 60 pooling attempts = 30 minutes. Is not it too much overhead here?